cli/common/src/snapshot.rs

```diff
@@ -260,16 +257,9 @@ impl<'a> Snapshot<'a> {
         self.get_bincode(&sysvar::stake_history::id())
     }
 
-    /// Read `sysvar::recent_blockhashes`.
-    pub fn get_recent_blockhashes(&mut self) -> crate::Result<RecentBlockhashes> {
-        self.get_bincode(&sysvar::recent_blockhashes::id())
-    }
-
     /// Return the most recent block hash at the time of the snapshot.
-    pub fn get_recent_blockhash(&mut self) -> crate::Result<Hash> {
-        let blockhashes = self.get_recent_blockhashes()?;
-        // The blockhashes are ordered from most recent to least recent.
-        Ok(blockhashes[0].blockhash)
+    pub fn get_latest_blockhash(&mut self) -> crate::Result<Hash> {
+        Ok(self.rpc_client.get_latest_blockhash()?)
```

Comment:
The point of the snapshot is that you can consistently read many things from a single atomic snapshot of the chain state. That means that the snapshot is not allowed to do any RPC calls, except for fetching the snapshot itself. If we make a new RPC call, then the blockhash that call returns can be a different block hash than the one the snapshot was built from.

(I recall something about Solana wanting to remove this sysvar. I also recall something about it being available in the raw RPC response. But not sure if my memory is correct here.)

Reply:
So maybe we should cache result of get_latest_blockhash() and reuse on new sign_transaction() calls?

Reply:
Done https://github.com/lidofinance/solido/blob/b07f2b32475850d6e167fb1f64620d57e0e58964/cli/common/src/snapshot.rs#L277-L289